crates/zk-por-core/src/circuits/merkle_sum_circuit.rs

```diff
@@ -0,0 +1,124 @@
+    hash::{
+        hash_types::HashOutTarget,
+        poseidon::PoseidonHash,
+    },
+    iop::target::Target,
+    plonk::circuit_builder::CircuitBuilder,
+};
+
+use crate::types::{F,D};
+
+use super::{account_circuit::AccountSumTargets, circuit_utils::assert_non_negative_unsigned};
+
+/// A node in the merkle sum tree, contains the total amount of equity (in usd) and the total amount of debt (in usd) and the hash.
+/// 
+/// The hash is Hash(hash_left, hash_right).
+/// 
+/// The amount of equity and amount of debt is the sum of the equity and debt of the children.
+#[derive(Debug, Copy, Clone,)]
+pub struct MerkleSumNodeTarget{
+    pub sum_assets: Target,
+    pub sum_debt: Target,
+    pub hash: HashOutTarget
+}
+
+impl MerkleSumNodeTarget {
+    /// Given two children, generate the next MerkleSumNode
+    pub fn get_child_from_parents(
+        builder: &mut CircuitBuilder<F, D>,
+        left_node: &MerkleSumNodeTarget,
+        right_node: &MerkleSumNodeTarget,
+    )->MerkleSumNodeTarget{
+        let sum_assets = builder.add(left_node.sum_assets, right_node.sum_assets);
+        let sum_debt = builder.add(left_node.sum_debt, right_node.sum_debt);
+
+        // Ensure the amount of assets at this node is greater than the total amount of debt
+        let diff_between_asset_debt = builder.sub(sum_assets, sum_debt);
+        assert_non_negative_unsigned(builder, diff_between_asset_debt);
+
+        let hash_inputs = vec![
+            left_node.hash.elements.to_vec(),
+            right_node.hash.elements.to_vec(),
+        ].concat();
+
+        let hash = builder.hash_n_to_hash_no_pad::<PoseidonHash>(hash_inputs);
+        MerkleSumNodeTarget{
+            sum_assets,
+            sum_debt,
+            hash
+        }
+    }
+
+    /// Get a merkle sum node from account sum targets.
+    pub fn get_node_from_account_targets(
+        builder: &mut CircuitBuilder<F, D>,
+        account_targets: &AccountSumTargets
+    )-> MerkleSumNodeTarget{
+        let hash_inputs = vec![
+            account_targets.sum_assets,
+            account_targets.sum_debt
+        ];
+
+        let hash = builder.hash_n_to_hash_no_pad::<PoseidonHash>(hash_inputs);
+        MerkleSumNodeTarget{
+            sum_assets: account_targets.sum_assets,
+            sum_debt: account_targets.sum_debt,
+            hash
+        }
+    }
+}
+
+
+/// We can represent the Merkle Sum Tree as a vector of merkle sum nodes, with the root being the last node in the vector.    
+pub struct MerkleSumTreeTarget{
+    pub sum_tree: Vec<MerkleSumNodeTarget>
+}
+
+impl MerkleSumTreeTarget{
+    pub fn get_root(&self)-> &MerkleSumNodeTarget{
+        self.sum_tree.last().unwrap()
+    }
+}
+
+/// Builds a merkle sum tree of a given size (based on the number of leaves). It will build the merkle sum tree on top of the leaves vector
+/// in order to do the task in place. There is no return value as the input leaves vector is mutated.
+pub fn build_merkle_sum_tree(
```

Comment:
need to handle leaves length not a strict power of 2

Reply:
Will be doing that in the next iteration, with the field merkle trees implementation , for this version I assume a power of two (this can easily be changed with padding but is not necessary since field merkle trees will deal with this issue).